src/main/java/com/artipie/management/Users.java

```diff
@@ -347,7 +169,8 @@ public boolean equals(final Object other) {
                 final User user = (User) other;
                 res = Objects.equals(this.uname, user.uname)
                     && Objects.equals(this.mail, user.mail)
-                    && Objects.equals(this.ugroups, user.ugroups);
+                    && this.ugroups.size() == user.ugroups.size()
+                    && this.ugroups.containsAll(user.ugroups);
```

Comment:
@olenagerasimova could you please explain why these changes are needed? They are possibly making `equals` and `hashCode` inconsistent. If this is done for tests, then it should be test-specific code

Reply:
@olegmoz groups order does not really matter, that's why I check the size and that one collection contains all elements as other

Reply:
@olenagerasimova the problem is that order might matter in `hashCode` calculation code. I suggest to change `ugroups` type to `Set` and revert code in `equals` method. Current version prone to breaking `equals/hashCode` contract